chain-connect/src/GalachainConnectClient.ts

```diff
@@ -103,3 +148,145 @@ export class GalachainConnectClient {
     return this.calculatePersonalSignPrefix(newPayload);
   }
 }
+
+export class PublicKeyClient {
+  constructor(
+    private client: GalachainConnectClient,
+    private url: string
+  ) {}
+
+  public updateUrl(url: string) {
+    this.url = url;
+  }
```

Comment:
```suggestion
  constructor(
    private readonly client: GalachainConnectClient,
    private readonly url: string
  ) {}
```

I think it is better to keep it simple, with no mutations.